src/components/content.js

```diff
@@ -718,9 +724,27 @@ class Content extends React.Component {
   render = () => {
     const { props } = this
     const { className, readOnly, state, tabIndex, role } = props
-    const { document } = state
+    const { document, selection } = state
+
+    let selectionNeedsRedraw = true
+
+    // If document and selection are the same we have stored after `onSelect`
+    // event, native selection does not need to be rendered again.
+    if (this.onSelectSelection) {
+      selectionNeedsRedraw = (
+        this.onSelectSelection.anchorKey !== selection.anchorKey ||
+        this.onSelectSelection.anchorOffset !== selection.anchorOffset ||
+        this.onSelectSelection.focusKey !== selection.focusKey ||
+        this.onSelectSelection.focusOffset !== selection.focusOffset ||
```

Comment:
Why the need here to compare all of the properties individually, instead of just comparing `onSelectSelection !== selection`?

If we really do need the deep comparison, we could use `Immutable.is` maybe?